test/test_prior.py

```diff
@@ -13,85 +14,101 @@
 from pypesto.objective.priors import get_parameter_prior_dict
 
+scales = ['lin', 'log', 'log10']
 
-def test_mode():
+
+@pytest.fixture(params=scales)
+def scale(request):
+    return request.param
+
+
+prior_types = ['uniform', 'normal', 'laplace', 'logNormal',
+               'parameterScaleUniform', 'parameterScaleNormal',
+               'parameterScaleLaplace']
+
+
+@pytest.fixture(params=prior_types)
+def prior_type(request):
+    return request.param
+
+
+def test_mode(scale, prior_type):
     """
     Tests the maximum/optimum for priors in different scales...
     """
 
-    scales = ['lin', 'log', 'log10']
-    prior_types = ['normal', 'laplace',
-                   'logNormal']
-
     problem_dict = {'lin': {'lb': [0], 'ub': [10], 'opt': [1]},
                     'log': {'lb': [-3], 'ub': [3], 'opt': [0]},
                     'log10': {'lb': [-3], 'ub': [2], 'opt': [0]}}
 
-    for prior_type, scale in itertools.product(prior_types, scales):
+    prior_list = [get_parameter_prior_dict(
+        0, prior_type, [1, 1], scale)]
 
-        prior_list = [get_parameter_prior_dict(
-            0, prior_type, [1, 1], scale)]
+    test_prior = NegLogParameterPriors(prior_list)
+    test_problem = pypesto.Problem(test_prior,
+                                   lb=problem_dict[scale]['lb'],
+                                   ub=problem_dict[scale]['ub'],
+                                   dim_full=1,
+                                   x_scales=[scale])
 
-        test_prior = NegLogParameterPriors(prior_list)
-        test_problem = pypesto.Problem(test_prior,
-                                       lb=problem_dict[scale]['lb'],
-                                       ub=problem_dict[scale]['ub'],
-                                       dim_full=1,
-                                       x_scales=[scale])
+    if prior_type.startswith('parameterScale'):
+        scale = 'lin'
 
-        optimizer = pypesto.optimize.ScipyOptimizer(method='Nelder-Mead')
+    optimizer = pypesto.optimize.ScipyOptimizer(method='Nelder-Mead')
 
-        result = pypesto.optimize.minimize(
-            problem=test_problem, optimizer=optimizer, n_starts=10)
-
-        assert np.isclose(result.optimize_result.list[0]['x'],
-                          problem_dict[scale]['opt'], atol=1e-04)
+    result = pypesto.optimize.minimize(
+        problem=test_problem, optimizer=optimizer, n_starts=10)
 
     # test uniform distribution:
-    for scale in scales:
-        prior_dict = get_parameter_prior_dict(
-            0, 'uniform', [1, 2], scale)
+    if prior_type in ['uniform', 'parameterScaleUniform']:
+
+        if prior_type == 'parameterScaleUniform':
+            scale = 'lin'
 
         # check inside and outside of interval
-        assert abs(prior_dict['density_fun'](lin_to_scaled(.5, scale))
-                   - 0) < 1e-8
+        assert abs(prior_list[0]['density_fun'](
+            lin_to_scaled(.5, scale)) - 0
```

Comment:
```suggestion
            lin_to_scaled(.5, scale))
```
Are these `- 0` or `- math.log(1)` from the old code intended?

Reply:
best to check via "diff < ...".

Reply:
Not sure what you are suggesting here.